### PR TITLE
issue #3576 - remove unnecessary guard against null resourceId

### PR DESCRIPTION
**Pull request by lmsurpre**
Now that our persistence layer requires resources to have proper
metadata (id, versionId, and lastUpdated) prior to calling create/update/delete, I thought that resources
will always have an id before we call the persistence interceptors.

However, it turns out that we call the beforeCreate from the REST layer
BEFORE setting this metadata on the resource.
After a brief discussion, we decided to leave that as-is and update the
interceptor to allow a null resourceId.

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java [hunk 1]
```diff
@@ -195,7 +195,7 @@ public FHIRRestHelper(FHIRPersistence persistence, SearchHelper searchHelper) {
 
     @Override
     public FHIRRestOperationResponse doCreate(String type, Resource resource, String ifNoneExist,
-        boolean doValidation) throws Exception {
+            boolean doValidation) throws Exception {
```
Comment (lmsurpre) [inline fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java:198]: unrelated formatting changes i had in my workspace

File: fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java [hunk 2]
```diff
@@ -954,7 +957,7 @@ private boolean isInCompartment(String resourceType, String resourceId, Resource
             for (String searchParmCode : inclusionCriteria) {
                 try {
                     SearchParameter inclusionParm = searchHelper.getSearchParameter(resourceType, searchParmCode);
-                    if (inclusionParm != null & inclusionParm.getExpression() != null) {
```
Comment (punktilious) [inline fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java:957]: ouch

**Review by punktilious — APPROVED** [review]
LGTM
